### DISCUSSION TITLE
E-book

### DISCUSSION THREAD

**Discussion by Dragomir-Ivanov** [go101/go101#184] [General]
Is there any way we can read this lovely book on our e-readers? Thanks a ton?

**Comment by zigo101** [comment]
The book will be listed in kindle and apple store soon.

You can also get the latest version and run a local server, then visit http://localhost:55555/article/print-book101 to show the print version, then you can print it to a pdf in chrome or firefox. Please note that links are deactivated in firefox and the print preview is slow in chrome, about several minutes maybe.

**Comment by Dragomir-Ivanov** [comment]
Thank you! Let me try that.

**Comment by Dragomir-Ivanov** [comment]
Used `Calibre` for converthing the PDF to MOBI, but the source code looks awful. Will wait and buy from kindle store. Thanks.

**Comment by zigo101** [comment]
Thanks for support!

**Comment by zigo101** [comment]
@Dragomir-Ivanov 
I made an epub and uploaded a version to Kindle Direct Publish: https://www.amazon.com/dp/B07Q3HWZ98, but I don't know whether or not Amazon converted it well.
I tried it on one my friend's kindle. It is readable, but the comments in code are not aligned as the web version.

I have set its price as the lowest one. $2.99. It is very appreciated if you can help me check whether or not the auto-converted version by Amazon is readable.

**Comment by zigo101** [comment]
I borrowed a kindle and checked this book.
It looks the code snippets in the code are not very readable.
I will adjust it again.

**Comment by Dragomir-Ivanov** [comment]
Thanks a lot. Ping me if you still want some testing. I guess you can ask a little more than $2.99 :)

**Comment by zigo101** [comment]
The final price will be $9.99, which is the highest prices for the 70% royalty model.
It is weird that Amazon will only share 35% royalty when book prices are higher than $9.99.

**Comment by Dragomir-Ivanov** [comment]
Don't tell me that amazon gets 70% for just providing the store?!!

**Comment by zigo101** [comment]
It is that if the price is higher than $9.99, than Amazon will always get 65%, and if the price is lower than $10, than you can choose to get 70%.

**Comment by Dragomir-Ivanov** [comment]
This is insane! Is the formatting okay now?

**Comment by zigo101** [comment]
Hi Dragomir-Ivanov,

I adjusted the styles again, it looks most code snippets look good in the Amazon preview website.
However, I don't know why Kindle service still hasn't pushed the latest version to the kindle I borrowed.

I have uploaded the offline version here: https://github.com/go101/go101/releases
It is great if you can help me check if the .azw3 version is ok on real devices.

**Comment by zigo101** [comment]
I just contacted Amazon customer service, they managed to make the book updated on the borrowed kindle. It looks the code font is some small. Currently, I set the code font size as 6pt. It looks it can be set as 7 pt safely. A larger font size will make many code lines break, which will hurt the code readability much.

So I will update a new version with code font size as 7pt.

BTW, the read experience is best under the "compact" read mode.

**Comment by Dragomir-Ivanov** [comment]
Amazon doesn't support PayPal. Can I send you ten bucks directly via paypal, and you get 100% royalty.

**Comment by zigo101** [comment]
Eh, my paypal happens in limited status now. I'm still in resolving this problem.
Don't mind it. Just download it from the release page. :)

BTW, I just fixed some style problems and uploaded a new release version.

**Comment by Dragomir-Ivanov** [comment]
Thanks a ton! Will try it tomorrow. Will see if my Visa Electron gets accepted in Amazon. I was using PayPal for everything these days...

**Comment by zigo101** [comment]
HI friend, you really don't need to pay for it.
If you would, instead, you can help me by recommending this book to your friends, or help me improve this book by reporting some mistakes you will find in reading this book.

**Comment by Dragomir-Ivanov** [comment]
Every effort should be repaid. This is my personal view, and I already bought it from Amazon. However not a regular Amazon customer so can't make a review on their store right now. Will recommend to all my friends using Go. Will try it on the reader later today. Thanks again, mate!

**Comment by zigo101** [comment]
OK, thanks for support!

**Comment by zigo101** [comment]
As the work for preparing ebook format versions is done.
I will close this issue.

**Comment by Dragomir-Ivanov** [comment]
The `azw3` file looks just fine on my Kindle Paper White, including the code segments. Thanks!

**Comment by zigo101** [comment]
Glad it is fine on your Kindle. The font is still a little small and there are a few lines being wrapped, but I have tried my best to make code snippets acceptable for reading.
